akka-docs/src/test/java/jdocs/persistence/testkit/TestKitExamples.java

```diff
@@ -0,0 +1,193 @@
+ */
+
+package jdocs.persistence.testkit;
+
+import akka.actor.typed.ActorSystem;
+import akka.persistence.testkit.DeleteEvents;
+import akka.persistence.testkit.DeleteSnapshotByMeta;
+import akka.persistence.testkit.DeleteSnapshotsByCriteria;
+import akka.persistence.testkit.JournalOperation;
+import akka.persistence.testkit.PersistenceTestKitPlugin;
+import akka.persistence.testkit.ProcessingPolicy;
+import akka.persistence.testkit.ProcessingResult;
+import akka.persistence.testkit.ProcessingSuccess;
+import akka.persistence.testkit.ReadEvents;
+import akka.persistence.testkit.ReadSeqNum;
+import akka.persistence.testkit.ReadSnapshot;
+import akka.persistence.testkit.Reject;
+import akka.persistence.testkit.SnapshotOperation;
+import akka.persistence.testkit.StorageFailure;
+import akka.persistence.testkit.WriteEvents;
+import akka.persistence.testkit.WriteSnapshot;
+import akka.persistence.testkit.javadsl.PersistenceTestKit;
+import akka.persistence.typed.PersistenceId;
+import akka.persistence.typed.javadsl.CommandHandler;
+import akka.persistence.typed.javadsl.EventHandler;
+import akka.persistence.typed.javadsl.EventSourcedBehavior;
+import com.typesafe.config.ConfigFactory;
+import jdocs.AbstractJavaTest;
+import org.junit.Test;
+
+public class TestKitExamples {
+
+  // #testkit-typed-usecase
+  class SampleTest extends AbstractJavaTest {
+
+    ActorSystem<Cmd> system =
+        ActorSystem.create(
+            new YourPersistentBehavior(PersistenceId.ofUniqueId("some-id")),
+            "example",
+            PersistenceTestKitPlugin.getInstance()
+                .config()
+                .withFallback(ConfigFactory.defaultApplication()));
+
+    @Test
+    void test() {
+      PersistenceTestKit testKit = PersistenceTestKit.create(system);
+
+      Cmd cmd = new Cmd("data");
+      system.tell(cmd);
+      Evt expectedEventPersisted = new Evt(cmd.getData());
+
+      testKit.expectNextPersisted("your-persistence-id", expectedEventPersisted);
+      testKit.clearAll();
+    }
+  }
+
+  class YourPersistentBehavior extends EventSourcedBehavior<Cmd, Evt, EmptyState> {
+
+    public YourPersistentBehavior(PersistenceId persistenceId) {
+      super(persistenceId);
+    }
+
+    @Override
+    public EmptyState emptyState() {
+      // some state
+      return new EmptyState();
+    }
+
+    @Override
+    public CommandHandler<Cmd, Evt, EmptyState> commandHandler() {
+      return newCommandHandlerBuilder()
+          .forAnyState()
+          .onCommand(Cmd.class, command -> Effect().persist(new Evt(command.getData())))
+          .build();
+    }
+
+    @Override
+    public EventHandler<EmptyState, Evt> eventHandler() {
+      // some logic
+      return null;
+    }
+  }
+  // #testkit-typed-usecase
+
+  // #set-event-storage-policy
+  class SampleEventStoragePolicy implements ProcessingPolicy<JournalOperation> {
+
+    // you can use internal state, it need not to be thread safe
```

Comment:
```suggestion
    // you can use internal state, it does not need to be thread safe
```
(this text appears in some more places)

Reply:
corrected

Reply:
Hi @manonthegithub, I think the comment looks like the original (before @raboof's suggested change).